lib/samson/secrets/vault_kv_wrapper.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+require 'vault'
+
+# Add recursive listing which was rejected by vault-ruby see https://github.com/hashicorp/vault-ruby/pull/118
+# using a monkey patch so `vault_action :list_recursive` works in the backend
+Vault::KV.class_eval do
+  def list_recursive(path, root = true)
+    keys = original_list(path).flat_map do |p|
+      full = +"#{path}#{p}"
+      if full.end_with?("/")
+        original_list_recursive(full, false)
+      else
+        full
+      end
+    end
+    keys.each { |k| k.slice!(0, path.size) } if root
+    keys
+  end
+
+  # Limit scope to logical after subclass call
+  alias_method :original_list, :list
+  alias_method :original_list_recursive, :list_recursive
+end
```

Comment:
are these 2 list_recursive versions different ... if not, a module to prepend in both would be nice

Reply:
... and if not, call out the diff in the docs

Reply:
I'll extract it to a shared module 👍